### PR TITLE
Recents items with shared links should populate the shared link header helper

### PR DESCRIPTION
**Pull request by boxtcarpel**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @boxtcarpel has signed the CLA. Thanks for the pull request!

**Review by crousselle — APPROVED** [review]

File: BoxContentSDK/BoxContentSDK/Requests/BOXRecentItemsRequest.m [hunk 1]
```diff
@@ -68,10 +68,24 @@ - (void)performRequestWithCompletion:(BOXRecentItemsBlock)completionBlock
...
                 if ([itemJSON isKindOfClass:[NSDictionary class]]) {
                     BOXRecentItem *recentItem = [[BOXRecentItem alloc] initWithJSON:itemJSON];
                     [mutableRecentItems addObject:recentItem];
+
+                    // Recent items could have been accessed on another end-point with a shared link.
+                    // In that case, we need to update the shared link header helper with this information.
+                    if (recentItem.sharedLinkURL != nil) {
+                        //FIXME: (adempsey, 2/28/17): currently does not support passwords. Items with
+                        // password-protected shared links are not currently returned by the recents API.
+                        if ([sharedLinkHelper sharedLinkForItemID:recentItem.item.modelID itemType:recentItem.item.type] == nil) {
```
Comment (adempsey) [inline BoxContentSDK/BoxContentSDK/Requests/BOXRecentItemsRequest.m]: If we have an expired shared link saved, wouldn't we want to overwrite it if the recent item returns a new one?

Comment (boxtcarpel) [reply BoxContentSDK/BoxContentSDK/Requests/BOXRecentItemsRequest.m]: I was trying to be more defensive, but I think you're right, it's better to just pick it up "blindly"

File: BoxContentSDK/BoxContentSDK/Requests/BOXRecentItemsRequest.m [hunk 2]
```diff
@@ -68,10 +68,22 @@ - (void)performRequestWithCompletion:(BOXRecentItemsBlock)completionBlock
             NSArray *recentsJSON = JSONDictionary[BOXAPICollectionKeyEntries];
             NSMutableArray<BOXRecentItem *> *mutableRecentItems = [NSMutableArray arrayWithCapacity:recentsJSON.count];
 
+            BOXSharedLinkHeadersHelper *sharedLinkHelper = [[BOXContentClient defaultClient] sharedLinksHeaderHelper];
```
Comment (adempsey) [inline BoxContentSDK/BoxContentSDK/Requests/BOXRecentItemsRequest.m]: super minor - no need to access the `sharedLinkHelper` until we're in the block after `recentItem.sharedLinkURL != nil)`

**Review by adempsey — APPROVED** [review]

**Pull request merged**